File: app/src/main/java/movies/com/ui/SplashFragment.java
package movies.com.ui;

import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;

import movies.com.MovieModel;
import movies.com.dbqr.R;
import movies.com.model.Movie;


public class SplashFragment extends Fragment {
    private MovieModel movieModel;

    public static SplashFragment newInstance() {
        return new SplashFragment();
    }

    public SplashFragment() {
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        movieModel = ViewModelProviders.of(this).get(MovieModel.class);
        movieModel.getMovieList().observe(this, movie -> {
            showListScreen(movie);
        });
        return inflater.inflate(R.layout.fragment_splash,
                container, false);

    }

    private void showListScreen(ArrayList<Movie> movieList) {
        MovieListFragment newFragment = MovieListFragment.Companion.newInstance(movieList);
        Bundle args = new Bundle();
        args.putParcelableArrayList(MovieListFragment.Companion.getARG_LIST(),  movieList);
        newFragment.setArguments(args);
        FragmentTransaction transaction = getActivity().getSupportFragmentManager().beginTransaction();
        transaction.replace(R.id.fragment_container, newFragment);
        transaction.addToBackStack(null);
        transaction.commit();
    }


}

File: app/src/main/java/movies/com/ui/SplashFragment.kt
package movies.com.ui

import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import movies.com.MovieModel
import movies.com.dbqr.R
import movies.com.model.Movie
import java.util.*


class SplashFragment : Fragment() {
    private var movieModel: MovieModel? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {

        movieModel = ViewModelProviders.of(this).get(MovieModel::class.java)
        movieModel!!.getMovieList()!!.observe(this, Observer { movies -> showListScreen(movies) })
        return inflater.inflate(R.layout.fragment_splash,
                container, false)

    }

    private fun showListScreen(movieList: ArrayList<Movie>?) {
        val newFragment = MovieListFragment.newInstance(movieList!!)
        val args = Bundle()
        args.putParcelableArrayList(MovieListFragment.ARG_LIST, movieList)
        newFragment.arguments = args
        val transaction = activity!!.supportFragmentManager.beginTransaction()
        transaction.replace(R.id.fragment_container, newFragment)
        transaction.addToBackStack(null)
        transaction.commit()
    }

    companion object {

        fun newInstance(): SplashFragment {
            return SplashFragment()
        }
    }


}
